nullaway/src/main/java/com/uber/nullaway/GenericsCheck.java

```diff
@@ -0,0 +1,145 @@
+package com.uber.nullaway;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Attribute;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.HashSet;
+import java.util.List;
+
+public class GenericsCheck {
+
+  // check that type is a valid instantiation of its generic type
+  public static void checkInstantiatedType(
+      Type type, VisitorState state, Tree tree, NullAway currentState, Config config) {
```

Comment:
Which `Tree` needs to be passed here?  I think it's the `Tree` upon which we want to report the error.  We should be sure to document that, and rename the variable.  Also I think `analysis` is a better name for the `NullAway` object than `currentState`:
```suggestion
      Type type, VisitorState state, Tree errorTree, NullAway analysis, Config config) {
```